apps/mobile/src/components/Login/SignInBottomSheet.tsx

```diff
@@ -87,13 +87,46 @@ function SignInBottomSheet({ onQrCodePress, openManageWallet }: Props) {
 }
 
 function FarcasterBottomSheetRow() {
-  const { open: handleConnectFarcaster } = useLoginWithFarcaster();
+  const [isFarcasterLoading, setIsFarcasterLoading] = useState(false);
+  const [attemptReconnect, setAttemptReconnect] = useState(false);
+
+  const {
+    open: handleConnectFarcaster,
+    isSuccess,
+    isConnected,
+    isPolling,
+  } = useLoginWithFarcaster({
+    setIsFarcasterLoading,
+  });
+
+  const [appState, setAppState] = useState(AppState.currentState);
+
+  useEffect(() => {
+    const subscription = AppState.addEventListener('change', (nextAppState) => {
+      if (appState.match(/inactive|background/) && nextAppState === 'active') {
+        if (isPolling || (!isConnected && !isSuccess)) {
+          setAttemptReconnect(true);
+        }
```

Comment:
this logic makes sense in that if the connection did not go through, we attempt to reconnect from scratch.

_is there any chance the connection is actually successful_, but our app doesn't pick up on it yet by the time the user arrives back on our app? just wondering if we need to wait at all before immediately going into reconnection mode